From 94519bd3495fd4c49d5c2d09188d0a43d7ad3bf8 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Tue, 15 Jul 2025 14:54:55 -0700
Subject: [PATCH] Fix install for multilib

For the multilib case, the lib directory name can and will be overridden,
so hard-coding `lib` is wrong.

Upstream-Status: Pending
Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 CMakeLists.txt            | 4 ++--
 src/common/CMakeLists.txt | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc3f733..3e6fdbc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ PROJECT(nxp_iot_agent LANGUAGES C VERSION 6.3.0)
 
 #define some variables which will be used in different places
 set(exported_targets el2go_agent-targets)
-set(config_install_dir lib/cmake/el2go_agent)
+set(config_install_dir ${CMAKE_INSTALL_LIBDIR}/cmake/el2go_agent)
 set(input_config_file ${CMAKE_CURRENT_SOURCE_DIR}/el2go_agent-config.cmake.in)
 set(config_file ${CMAKE_CURRENT_BINARY_DIR}/el2go_agent-config.cmake)
 set(version_config_file ${CMAKE_CURRENT_BINARY_DIR}/el2go_agent-config-version.cmake)
@@ -345,7 +345,7 @@ IF(SSS_HAVE_HOST_LINUX_LIKE OR NXP_IOT_AGENT_HAVE_PSA_IMPL_SMW)
     INSTALL(
         TARGETS ${PROJECT_NAME}
         EXPORT ${exported_targets}
-        DESTINATION lib
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
 
     INSTALL(FILES ${NXP_IOT_AGENT_PUBLIC_FILES} DESTINATION ${agent_include_install_dir})
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index a2d6895..88bac02 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -98,6 +98,6 @@ ENDIF()
 
 IF(SSS_HAVE_HOST_LINUX_LIKE OR NXP_IOT_AGENT_HAVE_PSA_IMPL_SMW)
 
-    INSTALL(TARGETS ${PROJECT_NAME} EXPORT ${exported_targets} DESTINATION lib)
+    INSTALL(TARGETS ${PROJECT_NAME} EXPORT ${exported_targets} DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 ENDIF()
-- 
2.34.1

